src/solvers/lowering/byte_operators.cpp

```diff
@@ -180,7 +180,46 @@ static exprt unpack_rec(
       max_bytes,
       ns);
   }
-  else if(ns.follow(src.type()).id()==ID_struct)
+  else if(src.type().id() == ID_complex)
+  {
+    const complex_typet &complex_type = to_complex_type(src.type());
+    const typet &subtype = complex_type.subtype();
+
+    auto subtype_bits = pointer_offset_bits(subtype, ns);
+    CHECK_RETURN(subtype_bits.has_value());
+
+    exprt::operandst byte_operands;
+
+    exprt sub_real = unpack_rec(
+      complex_real_exprt(src),
+      little_endian,
+      mp_integer{0},
+      *subtype_bits / 8,
+      ns,
+      true);
+    byte_operands.insert(
+      byte_operands.end(),
+      sub_real.operands().begin(),
+      sub_real.operands().end());
+
+    exprt sub_imag = unpack_rec(
+      complex_imag_exprt(src),
+      little_endian,
+      mp_integer{0},
+      *subtype_bits / 8,
+      ns,
+      true);
+    byte_operands.insert(
+      byte_operands.end(),
+      sub_imag.operands().begin(),
+      sub_imag.operands().end());
```

Comment:
use move instead

Reply:
How should I use move here given that I need to append to a vector? (The first insert of course should use move as you suggest.)

Reply:
`std::move(sub_imag.operands().begin(), sub_imag.operands().end(), std::back_inserter(byte_operands));`

Reply:
I'm now using `std::make_move_iterator` as the `std::back_inserter` would result in repeated reallocations. I hope that's ok as well.

Reply:
Yes I think that's fine, otherwise you would have to do a `reserve` beforehand.